src/block_cache/block_cache.c

```diff
@@ -554,13 +516,12 @@ __wt_blkcache_remove(WT_SESSION_IMPL *session, const uint8_t *addr, size_t addr_
         if (blkcache_item->addr_size == addr_size && blkcache_item->fid == S2BT(session)->id &&
           memcmp(blkcache_item->addr, addr, addr_size) == 0) {
             TAILQ_REMOVE(&blkcache->hash[bucket], blkcache_item, hashq);
-            blkcache->bucket_metadata[bucket].bucket_num_data_blocks--;
-            blkcache->bucket_metadata[bucket].bucket_bytes_used -= blkcache_item->data_size;
             __blkcache_update_ref_histogram(session, blkcache_item, BLKCACHE_RM_FREE);
```

Comment:
Do we want to assert here that the reference count is zero?

Reply:
I don't feel strongly one way or the other; the `blkcache_item.ref_count` is only non-zero for a short period of time, while the I/O routines are processing the data, but if you think it's useful I'm happy to add it.

Pushed in 087d106db.